src/contracts/libraries/Snapshots.sol

```diff
@@ -127,4 +130,129 @@ library Snapshots {
             result.slot := add(keccak256(0, 0x20), pos)
         }
     }
+
+    /**
+     * @notice WithdrawalHistory SnapShotting used for calculating burned amounts for StrategyManager Strategies
+     * Does not default to WAD, as the value is always 0 if there are no snapshots.
+     */
+    struct WithdrawalHistory {
+        WithdrawalSnapshot[] _snapshots;
+    }
+
+    struct WithdrawalSnapshot {
+        uint32 _blockNumber;
+        uint224 _value;
+    }
```

Comment:
On this note, I don't think we should be including these `SafeCastUpgradable` calls in `Snapshots`. It's adding bytecode and boy that's at a huge premium rn :sweat:

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/881/commits/173c411ca2bc66853c4f417d09557a35dc5a4935 Implemented your suggestion for reusability. 

I used SafeCastUpgradeable in one call when casting uint256 to uint224 since thats used in the underlying storage. However, I've validated that `uint224` is plenty large enough to store cumulative withdrawals in the protocol so happy to remove this too.

Reply:
I think we should do an unsafe case. We're doing unsafe casts for `uint64` and i don't see this as a different case - as long as we've validated we can't pass values larger than that.

It saves us ~150 bytes which we really need

Reply:
[e804598](https://github.com/Layr-Labs/eigenlayer-contracts/pull/881/commits/e80459845c6b945779b79ba8ffccfb65e91d1f58)